### PR TITLE
Add section to easily migrate an entire folder of code into "Migrate from BuckleScript/Reason" page

### PR DESCRIPTION
**Pull request by MoOx**
Because I think it's useful (and safe as soon as you have versioned codebase.

### CODE REVIEW

File: pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx [hunk 1]
```diff
@@ -16,12 +16,25 @@ ReScript is a rebranding and cleanup of BuckleScript (since `v8.2.0`) & Reason (
...
+
+Please use this with caution. Migrating an entire codebase at once is an uncessary risk.
+It is recommended to migrate small portion of code to ensure correctness.
+That being said, for small codebases or some tiny folders that are safely versionned, you can try the following:
+
+```console
+for f in your-folder/**/*.re; do; node_modules/.bin/bsc -format $f > ${f%.re}.res && rm $f; done;
+```
+
+This command loop on the `.re` files from the `your-folder` folder, convert them with a proper `.res` extension, and ask you if you want to remove the previous file (check the newly created `.res` file first). If you are confident enough, you can add `-f` option to the `rm` command.
```
Comment (TomiS) [inline pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx:36]: Typos: This command loop**s** ... convert**s** ... and ask**s**

File: pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx [hunk 2]
```diff
@@ -16,12 +16,25 @@ ReScript is a rebranding and cleanup of BuckleScript (since `v8.2.0`) & Reason (
...
+- `node_modules/.bin/bsc -format MyFile.re > MyFile.res`.
+- If your new `MyFile.res` looks good, you can delete (or move) `MyFile.re` (otherwise you will have an error `Invalid bsconfig.json implementation and interface have different path names or different cases MyFile vs MyFile`).
 
 **That's it**! `MyFile.re` could be any `.re`, `.rei`, `.ml` and `.mli` file.
 
 Enjoy the improved experience!
 
+### Upgrade an Entire Folder
+
+Please use this with caution. Migrating an entire codebase at once is an uncessary risk.
```
Comment (TomiS) [inline pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx:28]: Typo: unnecessary

Comment (MoOx) [reply pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx]: Fixed, thanks!

File: pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx (hunk 1 above)
Comment (MoOx) [reply pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx]: Thanks again! Fixed!

**Comment by ryyppy** [comment]
might be even worth mentioning to check in the compiled JS files first (compiled from the original .re files) before doing the conversion, bc checking the JS diffs is more reliable than just checking the new ReScript code.

Or to say it differently: if your compiled output doesn't change when compiling the files with the new syntax, you can be 100% sure your conversion didn't break any code

**Comment by MoOx** [comment]
That's a very good idea. I just added some instructions to do this comparison, tell me if that 's ok!

File: pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx [hunk 3]
```diff
@@ -16,12 +16,29 @@ ReScript is a rebranding and cleanup of BuckleScript (since `v8.2.0`) & Reason (
...
 
 **That's it**! `MyFile.re` could be any `.re`, `.rei`, `.ml` and `.mli` file.
 
 Enjoy the improved experience!
 
+### Upgrade an Entire Folder
+
+Please use this with caution. Migrating an entire codebase at once is an unnecessary risk.
+It is recommended to migrate small portion of code to ensure correctness.
+That being said, for small codebases or some tiny folders that are safely versionned, you can try the following:
```
Comment (harigopal) [inline pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx:33]:
```suggestion
That being said, for small codebases or some tiny folders that are safely versioned, you can try the following:
```

**Comment by harigopal** [comment]
This is nice! I'd just finished converting [pupilfirst's ReasonML to ReScript](https://github.com/pupilfirst/pupilfirst/pull/540) and I'd written a small Ruby script to help automate the conversion.

One thing that could possibly be included in the _total conversion_ snippet - for it to match `.rei` files to convert them to `.resi`.

File: pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx (hunk 3 above)
Comment (MoOx) [reply pages/docs/manual/latest/migrate-from-bucklescript-reason.mdx]: Thanks for the fix

**Comment by MoOx** [comment]
@harigopal not sure that's very necessary, a simple copy/paste with a 2 characters changes is acceptable I guess :D It's a "complicated" already line of code!

**Comment by ryyppy** [comment]
I think i will merge this for now, but I will probably rephrase a few things later, if that is okay

**Comment by MoOx** [comment]
Feel free to rephrase :)

**Pull request merged**